specification/source/cdl.rst

```diff
@@ -1688,118 +1688,97 @@ Therefore, tools that process CDL can infer the following information:
 
-[Note that, for example
-for a controller with two output signals :math:`y_1` and :math:`y_2`,
-Brick seems to have no means to specify
-that :math:`y_1` ``controls`` a fan speed and :math:`y_2` ``controls``
-a heating valve, where ``controls`` is the Brick relationship.
-Therefore, we allow the ``brick_annotation`` to only be at the
-block level, but not at the level of instances of input or output
-connectors.
+.. code-block:: modelica
+
+  annotation (__cdl(semantic(metadataLanguage="<metadataLanguage> [version] [format]" "<metadata>")));
+  annotation (__Buildings(semantic(metadataLanguage="<metadataLanguage> [version] [format]" "<metadata>")));
 
-For example, the Brick specification
+* Annotations attached to the top-level block can also be used to define block level information about the semantic data. A top-level block is one that encompasses a particular sequence or a model. For example, in Brick 1.3, it can be used to define prefixes. Example:
 
 .. code-block:: modelica
 
-   soda_hall:flow_sensor_SODA1F1_VAV_AV a brick:Supply_Air_Flow_Sensor ;
-      bf:hasTag brick:Average ;
-      bf:isLocatedIn soda_hall:floor_1 .
+  annotation (__cdl(semantic(
+    metadataLanguage="Brick 1.3 text/turtle" "@prefix Brick: <https://brickschema.org/schema/Brick#> .
+                                              @prefix bldg: <urn:bldg/> . ",
+    metadataLanguage=" Project-Haystack 3.9.12 application/ld+json"
+        "{\"@context\": { \"ph\": \"https://project-haystack.org/def/ph/3.9.12#\",
+          \"phScience\": \"https://project-haystack.org/def/phScience/3.9.12#\",
+          \"phIoT\": \"https://project-haystack.org/def/phIoT/3.9.12#\",
+          \"rdf\": \"http://www.w3.org/1999/02/22-rdf-syntax-ns#\",
+          \"rdfs\": \"http://www.w3.org/2000/01/rdf-schema#\"}}")));
 
-can be declared in CDL as
+* Additionally, if there already exists a metadata model for a particular sequence, it can be referred to in the annotation of the top-level block declaration using the "url" keyword (use ``file:///<path/to/file``) if the semantic model is present in a local machine. Example:
 
 .. code-block:: modelica
 
-   annotation(__cdl(brick="soda_hall:flow_sensor_SODA1F1_VAV_AV a brick:Supply_Air_Flow_Sensor  ;
-     bf:hasTag brick:Average ;
-     bf:isLocatedIn soda_hall:floor_1 ."));
-
-]
+  annotation (__cdl(semantic(metadataLanguage="Brick 1.3 text/turtle" "url=file:///path/to/model.ttl",
+   metadataLanguage="Project-Haystack 3.9.12 application/ld+json" "url=https://url/to/model.json")));
 
-For Haystack, the ``__cdl_annotation`` is
+* Example of CDL semantic information is shown below. The metadataLanguage "Brick 1.3 text/turtle" has been defined earlier in a top-level block and hence can be used to include semantic information. Similarly, "Project-Haystack 3.9.12 application/ld+json" and "Text" have been defined earlier and that makes for a valid syntax
```

Comment:
Does it really need to be top-level or will the `metadataLanguage` be taken from the first instance of a block that is at a higher hierarchy than the current instance? For example, if A extends B, and B extends C, and A and B contain `metadataLanguage`, does A override B, or B override A, or is it an error if their `metadataLanguage` are not the same?

Reply:
@mwetter: This is a good point that you bring up that we must probably be explicit in the standard. I need to think a bit on this and get back to you.